File: update_quickeligibility.py
with open("src/components/QuickEligibility.tsx", "r", encoding="utf-8") as f:
    content = f.read()

content = content.replace('export default function QuickEligibility() {', 'export default function QuickEligibility({ homeLoanRate, balanceTransferRate }: { homeLoanRate?: string, balanceTransferRate?: string }) {')

content = content.replace('const recommendedBank = empType === "salaried" ? "Top Nationalised Banks @ 6.50%" : "Leading Private Banks @ 6.60%";', 
                          'const recommendedBank = empType === "salaried" ? `Top Nationalised Banks @ ${homeLoanRate || "6.50"}%` : `Leading Private Banks @ ${balanceTransferRate || "6.60"}%`;')

with open("src/components/QuickEligibility.tsx", "w", encoding="utf-8") as f:
    f.write(content)

File: update_quick_eligibility.js
const fs = require('fs');
let content = fs.readFileSync('src/components/QuickEligibility.tsx', 'utf8');

content = content.replace(
  'export default function QuickEligibility({ homeLoanRate, businessLoanRate }: { homeLoanRate?: string, businessLoanRate?: string }) {',
  'export default function QuickEligibility({ homeLoanRate, selfEmployedRate }: { homeLoanRate?: string, selfEmployedRate?: string }) {'
);

content = content.replace(
  'const recommendedBank = empType === "salaried" ? `Top Nationalised Banks @ ${homeLoanRate || "6.50"}%` : `Leading Private Banks @ ${businessLoanRate || "12.50"}%`;',
  'const recommendedBank = empType === "salaried" ? `Top Nationalised Banks @ ${homeLoanRate || "6.50"}%` : `Leading Private Banks @ ${selfEmployedRate || "7.25"}%`;'
);

content = content.replace(
  'const interestRate = empType === "salaried" ? (parseFloat(homeLoanRate || "6.50") / 100) / 12 : (parseFloat(businessLoanRate || "12.50") / 100) / 12;',
  'const interestRate = empType === "salaried" ? (parseFloat(homeLoanRate || "6.50") / 100) / 12 : (parseFloat(selfEmployedRate || "7.25") / 100) / 12;'
);

fs.writeFileSync('src/components/QuickEligibility.tsx', content);
console.log('Updated QuickEligibility with selfEmployedRate');
